### PR TITLE
Update some eclasses for EAPI 8

### PR DESCRIPTION
**Pull request by ulm**

### CODE REVIEW

File: eclass/autotools.eclass [hunk 1]
```diff
@@ -31,7 +31,7 @@ case ${EAPI} in
 		# Needed for eqawarn
 		inherit eutils
 		;;
-	7) ;;
+	7|8) ;;
```
Comment (SoapGentoo) [inline eclass/autotools.eclass:34]: this needs to be put on hold, there's some other issues that need to be resolved first

File: eclass/git-r3.eclass [hunk 2]
```diff
@@ -4,35 +4,31 @@
...
 EXPORT_FUNCTIONS src_unpack
 
 if [[ ! ${_GIT_R3} ]]; then
 
-PROPERTIES+=" live"
+if [[ ${EAPI} != [567] ]]; then
+	PROPERTIES="live"
+else
+	PROPERTIES+=" live"
+fi
```
Comment (SoapGentoo) [inline eclass/git-r3.eclass:26]: I mean, I get it, but is this really such a big win? not that you should remove it, but it increases cyclomatic complexity purely for a syntactic sugar?

File: eclass/gnuconfig.eclass [hunk 3]
```diff
@@ -17,7 +17,7 @@
 #
 
 case ${EAPI:-0} in
-	5|6|7)
+	5|6|7|8)
```
Comment (SoapGentoo) [inline eclass/gnuconfig.eclass:20]: needs signoff from @thesamesam

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: eclass/git-r3.eclass (hunk 2 above)
Comment (ulm) [reply eclass/git-r3.eclass:26]: It's mainly a reminder for removal of the old `+=` assignment later.

File: eclass/gnuconfig.eclass [hunk 4]
```diff
@@ -17,7 +17,7 @@
...
 		;;
 	*)
 		die "EAPI ${EAPI} is unsupported!"
```
Comment (thesamesam) [inline eclass/gnuconfig.eclass:23]: While here:
```suggestion
		die "EAPI ${EAPI:-0} is unsupported!"
```

**Review by thesamesam — APPROVED** [review]

File: eclass/git-r3.eclass (hunk 2 above)
Comment (ulm) [reply eclass/git-r3.eclass:26]: Dropped.

File: eclass/libtool.eclass [hunk 5]
```diff
@@ -18,8 +18,8 @@ if [[ -z ${_LIBTOOL_ECLASS} ]]; then
 _LIBTOOL_ECLASS=1
 
 case ${EAPI:-0} in
-	0|1|2|3|4|5|6) DEPEND=">=app-portage/elt-patches-20170815" ;;
-	7) BDEPEND=">=app-portage/elt-patches-20170815" ;;
+	5|6) DEPEND=">=app-portage/elt-patches-20170815" ;;
+	7|8) BDEPEND=">=app-portage/elt-patches-20170815" ;;
 	*) die "${ECLASS}: EAPI ${EAPI} not supported" ;;
 esac
```
[the remark this answers is not in the document]
Comment (ulm) [reply eclass/libtool.eclass:25]: That doesn't look it would need to be blocked on EAPI update.

[the remark this answers is not in the document]
Comment (thesamesam) [reply eclass/libtool.eclass:25]: Sure.

**Review by thesamesam — APPROVED** [review]

**Pull request closed without merging**